content/docs/guides/tips-n-tricks.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "Tips, Tricks and Gotchas"
+weight: 4
+description: |
+  Tips and tricks for small places.
+  How to write efficient embedded code and avoid common mistakes.
+---
+
+## Ensure Concurrency
+
+TinyGo code runs on a single core, in a single thread (think `GOMAXPROCS=1`).
+This means we almost always have less threads than goroutines.
+Since scheduling in TinyGo is [cooperative](https://en.wikipedia.org/wiki/Cooperative_multitasking), a goroutine that never does IO or other blocking calls (f.ex. `time.Sleep()`) will lock the single available thread only for itself and never allow other goroutines to execute. In such cases `runtime.Gosched()` shall be used to force scheduler update.
+
+```
+package main
+
+import "fmt"
+import "time"
+import "sync/atomic"
+import "runtime"
+
+func main() {
+
+    var ops uint64 = 0
+    for i := 0; i < 50; i++ {
+        go func() {
+            for {
+                atomic.AddUint64(&ops, 1)
+                runtime.Gosched()
+            }
+        }()
+    }
+
+    time.Sleep(time.Second)
+
+    opsFinal := atomic.LoadUint64(&ops)
+    fmt.Println("ops:", opsFinal)
+}
+```
+TODO adapt example to embedded case
```

Comment:
TODO is for writers, not for readers. I don't think we should be showing this TODO to users. If you really want to keep it, consider `<!-- TODO ... -->` (not sure whether this syntax works, but I would assume so)

That said, I expect this code to work as-is on most chips.